File: Contracts/Model/Product/Response/SupplierResponse.cs
namespace Contracts.Model.Product.Response;

public class SupplierResponse : ByBaseResponse
{
	public Guid Id { get; set; }
	public string Name { get; set; }
	public string? Email { get; set; }
	public string? Description { get; set; }
	public string? Phone { get; set; }
	public string? Address { get; set; }
	public ICollection<ProductsListResponse>? Products { get; set; }
}

File: ClientApp/src/app/interfaces/product/response/supplierResponse.ts
import { ByBaseResponse } from "../../by-base-response";
import { ProductsListResponse } from "./products-list-response";

export interface SupplierResponse extends ByBaseResponse {
	id: string;
	name: string;
	email: string | null;
	description: string | null;
	phone: string | null;
	address: string;
	products: ProductsListResponse[] | null;
}